versions/1.17.1-fabric/src/main/kotlin/gg/essential/universal/shader/MCShader.kt

```diff
@@ -173,21 +175,29 @@ internal class ShaderTransformer {
             replacements["gl_Color"] = "uc_FrontColor"
         }
 
-        fun replaceAttribute(needle: String, type: String, replacementName: String = "uc_" + needle.substringAfter("_"), replacement: String = replacementName) {
+        fun replaceAttribute(newAttributes: MutableList<Pair<String, String>>, needle: String, type: String, replacementName: String = "uc_" + needle.substringAfter("_"), replacement: String = replacementName) {
             if (needle in source) {
                 replacements[needle] = replacement
                 if (replacementName !in attributes) {
                     attributes.add(replacementName)
```

Comment:
Should probably have these in sorted order too.
Afaict MC doesn't actually use them at the moment. Like, at all (they use the names from the `VertexFormat` for their `glBindAttribLocation` calls), so not sure what exactly its purpose was supposed to be. But if they ever do, having those be sorted incorrectly could lead to an unpleasant debugging session.

Reply:
I considered that, but I noticed that it was a Set so I assumed you'd already determined that order didn't mater. What was your goal with the `replacementName !in attributes`? I don't really see the point of checking for duplicate attributes, the shader is probably broken in other ways if that's the case.

Reply:
> I considered that, but I noticed that it was a Set so I assumed you'd already determined that order didn't mater. 

Yeah, not sure why I did that but considering I hadn't considered attribute order in general (hence why this PR exists), I doubt there's a deeper reason to it.

> What was your goal with the `replacementName !in attributes`? I don't really see the point of checking for duplicate attributes, the shader is probably broken in other ways if that's the case.

No clue. Seeing how this is the only place that `attributes` is modified, and that `replacementName` is only ever called with effectively hard-coded values, afaict it's literally impossible for that to ever be `false` to begin, regardless of what the shader does.